src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPlugin.java

```diff
@@ -0,0 +1,218 @@
+  static final String ERROR_GET_SECRETS_FAILED =
+      "[AWSSecretsManagerPlugin] Was not able to either fetch or read the database credentials from AWS Secrets Manager. Ensure the correct secretId and region properties have been provided";
+  static final String SQLSTATE_ACCESS_ERROR = "28000";
+  static final LRUCache<Pair<String, Region>, Secret> SECRET_CACHE = new LRUCache<>(100);
+  private final IConnectionPlugin nextPlugin;
+  private final Log logger;
+  private final String secretId;
+  private final Region region;
+  private final SecretsManagerClient secretsManagerClient;
+  private final GetSecretValueRequest getSecretValueRequest;
+
+  public AWSSecretsManagerPlugin(
+      ICurrentConnectionProvider currentConnectionProvider,
+      PropertySet propertySet,
+      IConnectionPlugin nextPlugin,
+      Log logger) throws SQLException {
+
+    this(
+        currentConnectionProvider,
+        propertySet,
+        nextPlugin,
+        logger,
+        null,
+        null
+    );
+  }
+
+  AWSSecretsManagerPlugin(
+      ICurrentConnectionProvider currentConnectionProvider,
+      PropertySet propertySet,
+      IConnectionPlugin nextPlugin,
+      Log logger,
+      SecretsManagerClient secretsManagerClient,
+      GetSecretValueRequest getSecretValueRequest) throws SQLException {
+
+    try {
+      Class.forName("software.amazon.awssdk.services.secretsmanager.SecretsManagerClient");
+    } catch (ClassNotFoundException e) {
+      logger.logError(ERROR_MISSING_DEPENDENCY);
+      throw new SQLException(ERROR_MISSING_DEPENDENCY);
+    }
+
+    this.nextPlugin = nextPlugin;
+    this.logger = logger;
+    this.secretId = propertySet.getStringProperty(SECRET_ID_PROPERTY).getValue();
+    this.region = Region.of(propertySet.getStringProperty(REGION_PROPERTY).getValue());
+
+    if (secretsManagerClient != null && getSecretValueRequest != null) {
+      this.secretsManagerClient = secretsManagerClient;
+      this.getSecretValueRequest = getSecretValueRequest;
+
+    } else {
+      this.secretsManagerClient = SecretsManagerClient.builder()
+          .region(this.region)
+          .build();
+      this.getSecretValueRequest = GetSecretValueRequest.builder()
+          .secretId(this.secretId)
+          .build();
+    }
+  }
+
+  @Override
+  public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLException {
+    final Properties updatedProperties = new Properties();
+    updatedProperties.putAll(connectionUrl.getOriginalProperties());
+    final Pair<String, Region> secretKey = Pair.of(secretId, region);
+    Secret secret = SECRET_CACHE.get(secretKey);
+    ConnectionUrl newConnectionUrl;
+
+    try {
+      // Attempt to open initial connection with cached secret
+      updateConnectionProperties(updatedProperties, secret);
+      newConnectionUrl = ConnectionUrl.getConnectionUrlInstance(connectionUrl.getDatabaseUrl(), updatedProperties);
```

Comment:
These two lines don't throw SQLException, so they can be taken out of the try catch clause